ansto_simplon_api/routes/detector/config.py

```diff
@@ -37,101 +77,312 @@ async def get_compression():
+# @router.put("/countrate_correction_count_cutoff")
+@router.get("/countrate_correction_count_cutoff")
+async def get_countrate_correction_count_cutoff():
+    return {"value": 133343}
 
 
-@router.get("/eiger_fw_version")
-async def get_eiger_fw_version():
-    return {"value": "release-2020.2.5"}
+# data_collection_date
 
 
-@router.get("/software_version")
-async def get_software_version():
-    return {"value": "1.8.0"}
+@router.put("/description")
+async def put_description(input: SimplonRequestStr):
+    zmq_start_message.detector_description = input.value
+    return {"value": zmq_start_message.detector_description}
 
 
 @router.get("/description")
 async def get_description():
-    return {"value": "Dectris EIGER2 Si 9M"}
+    return {"value": zmq_start_message.detector_description}
 
 
-@router.get("/detector_number")
-async def get_detector_number():
-    return {"value": "E-18-0108"}
+@router.get("/detector_distance")
+async def get_detector_distance():
+    return {"value": zmq_start_message.detector_translation}
 
 
-@router.get("/detector_type")
-async def get_detector_type():
-    return {"value": "HPC"}
+@router.put("/detector_distance")
+async def put_detector_distance(
+    input: tuple[
+        SimplonRequestFloat,
+        SimplonRequestFloat,
+        SimplonRequestFloat,
+    ]
+):
+    zmq_start_message.detector_translation = input.value
+    return {"value": zmq_start_message.detector_translation}
+
+
+@router.put("/detector_number")
+async def put_detector_number(input: SimplonRequestStr):
+    zmq_start_message.detector_serial_number = input.value
+    return {"value": zmq_start_message.detector_serial_number}
+
+
+@router.get("/detector_number")
+async def get_detector_number():
+    return {"value": zmq_start_message.detector_serial_number}
 
 
 @router.get("/detector_readout_time")
 async def get_detector_readout_time():
-    return {"value": 0.0000001}
+    return {"value": detector_configuration.detector_readout_time}
 
 
-@router.get("/bit_depth_image")
-async def get_bit_depth_image():
-    return {"value": 32}
+@router.put("/detector_readout_time")
+async def put_detector_readout_time(input: SimplonRequestFloat):
+    detector_configuration.detector_readout_time = input.value
+    return {"value": detector_configuration.detector_readout_time}
 
 
-@router.get("/bit_depth_readout")
-async def get_bit_depth_readout():
-    return {"value": 16}
+# @router.put("/eiger_fw_version")
+@router.get("/eiger_fw_version")
+async def get_eiger_fw_version():
+    return {"value": "release-2020.2.5"}
 
 
-@router.get("/countrate_correction_count_cutoff")
-async def get_countrate_correction_count_cutoff():
-    return {"value": 133343}
+# element
+# flatfield
+# flatfield_correction_applied
 
 
+# @router.put("/frame_count_time")
 @router.get("/frame_count_time")
 async def get_frame_count_time():
     return {"value": 0.004170816650000}
 
 
-@router.get("/number_of_excluded_pixels")
+@router.set("/frame_time")
```

Comment:
This is the last bug I found. `router.set`  should be replaced with `router.get`